### PR TITLE
Refactoring to use Bloc/Cubit state management.

### PR DESCRIPTION
**Pull request by susanto-tm**
Refactored models and state management system to use Blocs and Cubits and handle API Requests through Repositories.

### CODE REVIEW

**Comment by susanto-tm** [comment]
Resolved merge conflicts with main. Some issues with the CountdownTimer having memory leaks (potentially fixed) and refactored some of the userInfo fetching with the new Cubit/Repository pattern.

File: lib/utils/bloc/auth/auth_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  void _onAuthUserChanged(AuthUserChanged event, Emitter<AuthState> emit) {
+    emit(
+      event.user != null
+          ? AuthState.authenticated(event.user)
+          : AuthState.unauthenticated(),
+    );
+  }
+
+  void _onAuthLogout(AuthLogout event, Emitter<AuthState> emit) {
+    unawaited(_authenticationRepository.signOut());
```
Comment (rsquared226) [inline lib/utils/bloc/auth/auth_bloc.dart:38]: `"The method 'unawaited' isn't defined for the type 'AuthBloc'.\nTry correcting the name to the name of an existing method, or defining a method named 'unawaited'."`
I looked it up and found this: https://stackoverflow.com/a/70733424
What can we do about it?

**Review by rsquared226 — COMMENTED** [review]
I skimmed through all the changes and they look pretty good! I can't run it yet though because of a small error, but I think that should be easy to fix, I left it as a line comment.

Also when learning Flutter I learned that it was better practice to do relational imports like `import './utils/flavor_constants.dart';` for our own files rather than things like `import 'package:hackpsu/card_items/countdown_timer_card.dart';`, but I think what's more important is consistency. I can change all the imports to be more consistent, but what do you think about that?

File: lib/utils/bloc/auth/auth_bloc.dart (hunk 1 above)
Comment (susanto-tm) [reply lib/utils/bloc/auth/auth_bloc.dart:38]: I think I can just remove it. I forgot that's for a future version of Dart.

**Review by rsquared226 — APPROVED** [review]
Looks AMAZING to me! Just did a couple super small things in my commit but other than that I think it's good to go! Feel free to merge if you think it's good!

**Pull request merged**